app/controllers/api/v1/recordings_controller.rb

```diff
@@ -57,19 +57,28 @@ def resync
         render_json(status: :ok)
       end
 
-      def publish_recording
-        publish = params[:publish]
+      # POST /api/v1/recordings/update_visibility.json
+      def update_visibility
         record_id = params[:record_id]
-        case publish
-        when 'true'
-          visibility = 'Published'
-        when 'false'
-          visibility = 'Unpublished'
+        visibility = params[:visibility]
+
+        case visibility
+        when 'Published'
```

Comment:
I think this can be given to the `#publish_recordings` which will have the complexity of this business logic to determine the right configuration and maybe call `#update_recordings` or other services when needed.

Reply:
I'll put some thoughts into trying to optimize this.